website/src/views/modules/ModuleFinderSidebar.tsx

```diff
@@ -40,6 +48,37 @@ const EXAM_FILTER_ITEMS: FilterItem[] = [
 
 const ModuleFinderSidebar: React.FC = React.memo(() => {
   const [isMenuOpen, setMenuOpen] = useState(false);
+
+  const selectedModules = useSelector((state: StoreState) => {
+    const {
+      app: { activeSemester },
+      moduleBank: { modules },
+      timetables,
+    } = state;
+    const { timetable } = getSemesterTimetable(activeSemester, timetables);
+    const allSemesterModules = getSemesterModules(timetable, modules);
+    return allSemesterModules.map((module) => getModuleSemesterData(module, activeSemester));
+  });
+
+  const generateExamDateClashFilter: () => FilterItem[] = () => {
+    const examDates = selectedModules.map((module) => {
+      return module?.examDate;
+    });
+
+    return [
+      {
+        key: 'no-exam-clash',
+        label: 'No Exam Clash with Currently Selected Modules',
+        filter: BoolMustNot(TermsQuery('semesterData.examDate', examDates)),
+      },
+    ];
+  };
+
+  const getExamFilters: () => FilterItem[] = () => {
+    const examDateClashFilter = generateExamDateClashFilter();
+    return EXAM_FILTER_ITEMS.concat(examDateClashFilter);
+  };
```

Comment:
This and `generateExamDateClashFilter` don't seem to need to be functions. You should be able to inline `generateExamDateClashFilter` here and use `useMemo` 

```suggestion
  const examFilters = useMemo(() => {
    const examDates = selectedModules.map((module) => module?.examDate);
    const examDateClashFilter = {
      key: 'no-exam-clash',
      label: 'No Exam Clash with Currently Selected Modules',
      filter: BoolMustNot(TermsQuery('semesterData.examDate', examDates)),
    };
    
    return [...EXAM_FILTER_ITEMS, examDateClashFilter];
  }, [selectedModules]);
```